### ISSUE TITLE
get allele seqs (and/or SNPs) from sgc GFA file?

### ISSUE THREAD

**Issue by maureenbug** [spacegraphcats/spacegraphcats#446]
hi!

I want to pull all the different "alleles" (the set of unique sequences you'd get if you follow the paths along the graph) from the GFA file (mainly to look for SNPs, but I also want the actual sequences for some other stuff too). I put together an (inefficient and clumsy, attached) python script to output all the sequences from a given graph.

It works fine, but gets really slow on large graphs, especially when I'm trying to get all the paths I want from the GFA (networkx will give every single path in the graph, but most of those paths will be subsets of a larger/longer path... and I don't necessarily want those).

I was wondering if 1) there was at least a way for sgc to output a file with all the unique paths (without any shorter paths that are subsets of the larger paths), and/or 2) a way for sgc to actually output a fasta file with the "allele" sequences?

thanks!
maureen

[gfa_get_allele_seqs_sgc.txt](https://github.com/spacegraphcats/spacegraphcats/files/7420275/gfa_get_allele_seqs_sgc.txt)

**Comment by ctb** [comment]
great question, @maureenbug!

The simplest thing I can think of right now would be to get the individual reads that match to the neighborhood of the sequence in question (this is straightforward!) and then run a variant calling pipeline on that. Would this give you what you want? If so I can maybe put something together.

**Comment by maureenbug** [comment]
yeah I think this would get me what I want! would be great if you had suggestions on how to do the readmapping for this!

**Comment by ctb** [comment]
hi @maureenbug sorry I lost track of this during what was a pretty terrible November, but! I implemented something that might get you closer to what you want - 

https://github.com/spacegraphcats/spacegraphcats/pull/449#issuecomment-981656497

In brief, you can use this to get a minimum collection of reads that "cover" the k-mers in your region of interest. It won't run on really large graphs without a lot of time/memory, though.
